src/main/java/gov/cabinetoffice/gap/adminbackend/services/OpenSearchService.java

```diff
@@ -0,0 +1,66 @@
+package gov.cabinetoffice.gap.adminbackend.services;
+
+import com.contentful.java.cma.model.CMAEntry;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import gov.cabinetoffice.gap.adminbackend.config.OpenSearchConfig;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+import static org.apache.http.HttpHeaders.AUTHORIZATION;
+import static org.apache.http.HttpHeaders.CONTENT_TYPE;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class OpenSearchService {
+
+    private final WebClient.Builder webClientBuilder;
+    private final OpenSearchConfig openSearchConfig;
+    private final ObjectMapper objectMapper;
+
+    public void indexEntry(final CMAEntry contentfulEntry) {
+        final String body = contentfulEntryToJsonString(contentfulEntry);
+        webClientBuilder.build().put()
+                .uri(createUrl(contentfulEntry))
+                .body(Mono.just(body), String.class)
+                .header(CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE + "; " + StandardCharsets.UTF_8.name())
+                .header(AUTHORIZATION, createAuthHeader())
+                .retrieve()
+                .bodyToMono(void.class)
+                .block();
```

Comment:
Can use `.doOnError()` to specify what we want to do if the request fails. think we should log it or retry